modules/qualitycontrol/CodecSelection.js

```diff
@@ -281,7 +204,43 @@ export class CodecSelection {
             return;
         }
 
-        session.setVideoCodecs(selectedCodecOrder);
+        session.setVideoCodecs(selectedCodecOrder, this.screenshareCodec?.jvb);
+    }
+
+    /**
+     * Changes the codec preference order when initiated by the {@link QualityController} to improve video quality.
+     *
+     * @param {JitsiLocalTrack} localTrack - The local video track for which stats are being processed.
+     * @param {CodecMimeType} codec - The codec used for encoding the given local video track.
+     * @returns boolean - Returns true if the codec order has been updated, false otherwise.
+     */
+    changeCodecPreferenceOrder(localTrack, codec) {
+        const session = this.conference.getActiveMediaSession();
+        const connectionType = session.isP2P ? 'p2p' : 'jvb';
+        const codecOrder = this.codecPreferenceOrder[connectionType];
+        const videoType = localTrack.getVideoType();
+        const codecsByVideoType = VIDEO_CODECS_BY_COMPLEXITY[videoType]
+            .filter(val => Boolean(codecOrder.find(supportedCodec => supportedCodec === val)));
+        const codecIndex = codecsByVideoType.findIndex(val => val === codec.toLowerCase());
+
+        if (codecIndex < codecsByVideoType.length - 1) {
+            const newCodec = codecsByVideoType[codecIndex + 1];
+
+            if (localTrack.getVideoType() === VideoType.CAMERA) {
+                const idx = codecOrder.findIndex(val => val === newCodec);
+
+                codecOrder.splice(idx, 1);
+                codecOrder.unshift(newCodec);
+            } else {
+                this.screenshareCodec[connectionType] = newCodec;
+            }
+
+            this.selectPreferredCodec(session);
+
+            return true;
+        }
+
+        return false;
```

Comment:
I think the code will look nicer to read if you reverse the check and return early, this way you'll reduce one lavel of indentation.